File: backend/Services/Parsers/BazingleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class BazingleParser : BasicScoreResultParser
    {
        private ILogger<BazingleParser> _logger;

        public BazingleParser(ILogger<BazingleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => false;
        public override string GameName => "Bazingle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{GameName} .+?(?<{ScoreGroup}>[\\d|X|x])/\\d");
        protected override string? ExtraContent => Url;
        public override string Url => "https://bazingle.wook.wtf";
    }
}

File: lambdas/src/parsers/bazingle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class BazingleParser extends BasicScoreResultParser {
  readonly countWinner = false;
  readonly gameName = 'Bazingle';
  readonly helpText = null;
  readonly url = 'https://bazingle.wook.wtf';
  protected readonly extraContent = this.url;
  
  // Matches "Bazingle 683 1/8"
  // Note: C# uses [\d|X|x] which incorrectly includes literal pipes - fixed here to [\dXx]
  protected readonly parser = new RegExp(
    `${this.gameName} .+?(?<${BasicScoreResultParser.SCORE_GROUP}>[\\dXx])/\\d`
  );
}
